software/tests/test_graphs.py

```diff
@@ -33,24 +32,28 @@
 # Note that known failings can be annotated with @unittest.expectedFailure or @skip("reason...")
 class SDOGraphSetupTestCase(unittest.TestCase):
 
-  @classmethod
-  def loadGraphs(self):
-      if not SdoTermSource.SOURCEGRAPH:
-        SdoTermSource.loadSourceGraph("default")
-      self.rdflib_data = SdoTermSource.sourceGraph()
+  log = logging.getLogger(__name__)
 
+  @classmethod
+  def loadGraphs(cls):
+    if not SdoTermSource.SOURCEGRAPH:
+      SdoTermSource.loadSourceGraph("default")
+    cls.rdflib_data = SdoTermSource.sourceGraph()
 
   @classmethod
-  def setUpClass(self):
-    log.info("Graph tests require rdflib.")
+  def setUpClass(cls):
+    cls.log.info("Graph tests require rdflib.")
     try:
-      log.info("Trying to import rdflib...")
+      cls.log.info("Trying to import rdflib...")
       import rdflib
       from rdflib import Graph
     except Exception as e:
       raise unittest.SkipTest("Need rdflib installed to do graph tests: %s" % e)
```

Comment:
should rdflib simply be required, instead of skipping tests entirely when it's not here?
It would simplify the code quite a bit, and ensure we run all the tests at each release.

Reply:
Probably, but given how the tests are entwined I would defer this until we fix the general importing logic.